proposed/http-client/http-client.md

```diff
@@ -3,16 +3,31 @@ HTTP Client
 
 This document describes common interfaces for sending HTTP messages.
 
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
 
 ## Specification
 
+### Client
+
 An HTTP client has the responsibility to send a PSR-7 request and return a PSR-7
-response. When there is an error during sending the request or an error with network an
-exception should be thrown. 
+response. The HTTP client MAY modify the request, given by the user and sent to 
+the server or the response from the server before returning. In this case the 
+request and the response must be consistent between the body and headers. I.e. a 
+server may return a gzip encoded body and the client may know how to decode this, 
+when it decodes the body the client MUST also remove the header that specifies this 
```

Comment:
using a `MUST` inside an example doesn't make much sense. Suggestion:
`and the client may know how to decode this, however the client should then also mutate the headers that specify the encoding`

Reply:
hm, the MUST is important here. should we move that out of the example to say that if the client modifies the request or response, it MUST also adjust the headers accordingly? (i'd assume that if it gzips a request, it must also adjust the content-length header, for example)

Reply:
👍